packages/e2e-tests/src/specs/dashboard/settings/adminUser/customFonts.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import {
+  withExperimentalFeatures,
+  visitSettings,
+  addCustomFont,
+  removeCustomFont,
+  getFontList,
+  getSelectedFontList,
+  takeSnapshot,
+} from '@web-stories-wp/e2e-test-utils';
+
+const FONT_BASE_URL = `${process.env.WP_BASE_URL}/wp-content/e2e-assets`;
+const OPEN_SANS_CONDENSED_LIGHT = 'Open Sans Condensed Light';
+const OPEN_SANS_CONDENSED_LIGHT_URL = `${FONT_BASE_URL}/OpenSansCondensed-Light.ttf`;
+const OPEN_SANS_CONDENSED_BOLD = 'Open Sans Condensed Bold';
+const OPEN_SANS_CONDENSED_BOLD_URL = `${FONT_BASE_URL}/OpenSansCondensed-Bold.ttf`;
+const OPEN_SANS_CONDENSED_LIGHT_ITALIC_URL = `${FONT_BASE_URL}/OpenSansCondensed-LightItalic.ttf`;
+
+const findByUrl = (arr, val) => {
+  const found = arr.filter((o) => o.url === val);
+  if (found) {
+    return found[0];
+  }
+  return [];
+};
+
+const asyncForEach = async (array, callback) => {
+  for (let index = 0; index < array.length; index++) {
+    // eslint-disable-next-line no-await-in-loop -- async foreach
+    await callback(array[index], index, array);
+  }
+};
+
+const removeAllFonts = async () => {
+  await visitSettings();
+  let fonts = await getFontList();
+
+  if (fonts.length === 0) {
+    return;
+  }
+
+  await asyncForEach(fonts, async (font) => {
+    await removeCustomFont(font.name);
```

Comment:
Instead of this could do. 

```
await Promise.all( fonts.map( ( { name } ) => removeCustomFont(name) ) );
```

Might be a little simplier to read / understand.

Reply:
```
await Promise.all( fonts.map( ( { name } ) => removeCustomFont(name) ) );
```

So this is an interesting one.

I was using promise.all map --- but the font deletion would sometimes happen and other times fail.

I would need to play around with it further but I did notice when capturing some screens that the "next" deletion looked like it was trying to fire before the first dialog box "delete font" click was happening.  In turn one or more fonts would still end up in the list.

With the async for each --- the code awaits properly + runs in order without the next delete firing pre the first one being finished.

Reply:
If they need to run it order, then just do a for loop and add an ignore to the jslint that comes up. No need for another function.

Reply:
Needs to be async --- need to wait for the previous font deleting to finish 

press delete 
font dialog -> delete

repeat

Reply:
@spacedmonkey your right for loop is all that's needed here - 

Tested here : https://codesandbox.io/s/reverent-fermi-h1vgex

Reply:
Pushed the update